### PR TITLE
Simple tags for simple grug

### PR DESCRIPTION
**Pull request by KMontag42**
loads tags from core CMS data
lets users select them from a dropdown in the save modal
displays tags in the formation card
lets users search by tags on the search page

https://discord.com/channels/731929330896338995/1293754624037097555/1301764154868109344

gave insiders a chance to take a look at this, hoping they'll give us some good feedback and feel involved in the process.

ideally i want to ship by the weekend :)

### CODE REVIEW

File: app/api/formations/[id]/route.ts [hunk 1]
```diff
@@ -28,6 +28,8 @@ export async function PUT(
 
   await updateFormation(parseInt(id), data);
 
+  // clear the cache for the formation here
```
Comment (DMills94) [inline app/api/formations/[id]/route.ts:31]: is this a TODO?

File: components/Builder.tsx [hunk 2]
```diff
@@ -96,6 +96,8 @@ export default function Builder({ data, formation: _formation }: Props) {
     .sort((a, b) => a.name.localeCompare(b.name));
 
   const changeLayout = (newLayoutId: number) => {
+    console.log(newLayoutId);
+    console.log(layout);
```
Comment (DMills94) [inline components/Builder.tsx:100]: assume these are to be removed?

File: components/SaveButton.tsx [hunk 3]
```diff
@@ -131,7 +153,15 @@ function SaveFormationForm({
...
-}: React.ComponentProps<"form">) {
+  availableTags,
+  selectedTag,
+  setSelectedTag,
+}: React.ComponentProps<"form"> & {
+  availableTags: { name: string; value: string }[];
+  selectedTag: string | undefined;
+  setSelectedTag: (tag: string) => void;
+}) {
+  console.log(selectedTag);
```
Comment (DMills94) [inline components/SaveButton.tsx:164]: another

File: components/FormationCard.tsx [hunk 4]
```diff
@@ -55,7 +58,14 @@ export default function FormationCard({
   const cardHeaderAndContent = (
     <>
       <CardHeader>
-        <CardTitle>{name}</CardTitle>
+        <CardTitle>
+          <div className="flex flex-col">
+            <div>{name}</div>
```
Comment (DMills94) [inline components/FormationCard.tsx:63]: should this be a text html element like `p`?

File: components/Builder.tsx [hunk 5]
```diff
@@ -63,7 +63,7 @@ export default function Builder({ data, formation: _formation }: Props) {
   const [layout, setLayout] = useQueryState<number>("map", {
     parse: (query: string): number => parseInt(query),
     serialize: (layout: number) => layout.toString(),
-    defaultValue: _formation ? _formation.layout : 0,
+    defaultValue: _formation ? Math.trunc(_formation.layout) : 0,
```
Comment (DMills94) [inline components/Builder.tsx:66]: curious what this fixed?

File: components/Search.tsx [hunk 6]
```diff
@@ -30,14 +37,20 @@ export default function Search({
...
     e.preventDefault();
 
     setLoading(true);
 
-    const query = encodeURIComponent(search);
-    const response = await fetch(`/api/search?q=${query}`);
+    const params = new URLSearchParams();
+    params.append("q", search);
+    if (tag !== "") {
+      params.append("tag", tag);
```
Comment (SwiftieHusband) [inline components/Search.tsx:51]: Why not use just `t` here to follow the same convention as q for query? Doesn't really matter, just little details

**Review by SwiftieHusband — COMMENTED** [review]
Aside from my one comment and Baes, looks good. Question though, why only allow one tag to be selected in the select? I'm assuming we want the user to be able to select a few at the same time?

File: components/Builder.tsx (hunk 5 above)
Comment (KMontag42) [reply components/Builder.tsx:66]: this fixed a situation where some of our older formations had values like `0.0` for their `layout`.

by doing `Math.trunc` we ensure that the values are coerced into integers for use later on in the `layouts` lookup.

in the future this process will get phased out, but for now this ensures we don't hit any weird bugs.

**Comment by KMontag42** [comment]
> why only allow one tag to be selected in the select? I'm assuming we want the user to be able to select a few at the same time?

for now i wanted to see how people interacted with the system and build out the initial list of tags.

i left room in the code itself to support multiple tags (ie the field is a json array of strings) but it was going to add a bunch of complexity for something that will be new to our users.

File: app/api/formations/[id]/route.ts (hunk 1 above)
Comment (KMontag42) [reply app/api/formations/[id]/route.ts:31]: i had left this as a note for myself during some investigation, will remove. good catch!

**Comment by KMontag42** [comment]
> > why only allow one tag to be selected in the select? I'm assuming we want the user to be able to select a few at the same time?
> 
> for now i wanted to see how people interacted with the system and build out the initial list of tags.
> 
> i left room in the code itself to support multiple tags (ie the field is a json array of strings) but it was going to add a bunch of complexity for something that will be new to our users.

looks like the people want multiple tags!

i'll cook on this and some colors tonight

**Pull request merged**